File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int s = 0, e = n-1;
        while(s < e){
            int sum = nums[s] + nums[e];
            if(sum == target){
                return new int[]{s+1, e+1};
            }
            else if(sum < target){
                s++;
            }
            else{
                e--;
            }
        }
        return new int[]{-1, -1};
    }
}

File: Data Structures & Algorithms/car-fleet/submission-1.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        cars = sorted(zip(position, speed), reverse = True)
        slow_time, fleet = 0.0, 0
        #stack = []
        for pos, spd in cars:
            time = (target - pos)/spd
            if time > slow_time:
                fleet += 1
                slow_time = time
            # if not stack or time > stack[-1]:
            #     stack.append(time)
        return fleet

            